### ISSUE TITLE
Running socketcluster using pm2 or forever keep display blank node.exe cmd

### ISSUE THREAD

**Issue by bebeo92** [SocketCluster/socketcluster#440]
Hi I'm and my team try to develop an app using socketcluster and now want to deploy it using pm2 at begin

But when we try to deploy it using pm2 and latter I try to use forever it seem it faced same issue

Step to reproduce
Install latest socket cluster version using npm
Create new project or using code from master branch
Try to run sample project using pm2(or forever)
forever start --killSignal=SIGTERM server.js -w 2 -s 2
pm2 start server.js

![1](https://user-images.githubusercontent.com/9253836/44798630-35e2a200-abdc-11e8-943f-80bff196236e.png)

Is there any that I doing wrong, is there a way to work around it ? Or this is an issue

**Comment by bebeo92** [comment]
@jondubois Yes, if I kill main process, it will turn off that child process
But you know, when you deploy it into production, you should find a way to run it in background and that it nothing else (no blank screen....).
And in my case, our team using pm2, Imagine like you have hundred app running on server and each app need a console to display... it will be a mess right?

So my expectation is find out a way to deploy app using socketcluster using script and it will running silently in background. Is there a way to do it using pm2 or other tool that, using socketcluster by itself I would like to know?

**Comment by jondubois** [comment]
@bebeo92 it definitely doesn't do it on Linux (which is the most common production environment). Maybe PM2 is not optimized for Windows.

SC just uses node.js's `cluster` and `child_process` modules to spawn up child processes so it's not doing anything out of the ordinary. Have you tried using a different console in Windows? Maybe Powershell?

In any case, this is a PM2 issue, not a SocketCluster issue: https://github.com/Unitech/pm2/issues/2182

**Comment by bebeo92** [comment]
@jondubois Ok then so is there any easy way to start and stop SC automatically?
Sorry but I stick with window ....

**Comment by jondubois** [comment]
Otherwise, if you think that this issue can be solved inside SC, you can try playing around with SC's use of `fork`:

- In main `socketcluster` module: https://github.com/SocketCluster/socketcluster/blob/master/scworkercluster.js#L181
- In `sc-broker` dependency: https://github.com/SocketCluster/sc-broker/blob/master/index.js#L82

**Comment by bebeo92** [comment]
@jondubois pm2 already in latest version ...

Option 2 look promise then

**Comment by mihranharutyunyan** [comment]
@bebeo92 pm2 both for windows and for linux shows only the master process you start: server.js, it does not care of the child processes (worrkers, brokers, ...) and does not manage them.  You can see them in the process list
